### PR TITLE
Support domain-scoped project ID in the billing export table name

### PR DESCRIPTION
**Pull request by sarandia**
This is because GCP used to support domain-scoped project IDs in the
format of "abc.xyz:abcd".

### CODE REVIEW

File: gke-usage-metering/src/gkeUsageMetering.ts [hunk 1]
```diff
@@ -153,7 +153,9 @@ namespace gkeUsageMetering {
     }
     // fullBillingTable should be in the format of ${PROJECT_ID}.${DATASET_ID}.${TABLE_ID}.
     // Bigquery table IDs can contain letters, numbers and underscores.
-    if (!fullBillingTable.match(/^[a-z][a-z-\d]{5,29}\.[a-zA-Z_\d]{1,1024}\.[a-zA-Z_\d]{1,1024}$/)) {
```
[the remark this answers is not in the document]
Comment (sarandia) [reply gke-usage-metering/src/gkeUsageMetering.ts:156]: Done.

**Comment by jacobstr** [comment]
Having trouble connecting GKE usage metering - possibly related? Might be a keyboard-and-chair problem on my part but the datastudio connector is rejecting my submission with:

```
Invalid GCP billing table ID: table ID must be in the format of "${PROJECT_ID}.${DATASET_ID}.${TABLE_ID}"
```

I was able to create the connector successfully about a month ago.

**Comment by sarandia** [comment]
> Having trouble connecting GKE usage metering - possibly related? Might be a keyboard-and-chair problem on my part but the datastudio connector is rejecting my submission with:
> 
> ```
> Invalid GCP billing table ID: table ID must be in the format of "${PROJECT_ID}.${DATASET_ID}.${TABLE_ID}"
> ```
> 
> I was able to create the connector successfully about a month ago.

We recently changed the input requirement to no longer allow the project ID and dataset ID to be separated by ":". Could that be the reason?

File: gke-usage-metering/src/gkeUsageMetering.ts [hunk 2]
```diff
@@ -151,9 +151,15 @@ namespace gkeUsageMetering {
...
+    // We support two types of project IDs: the current standard, and the previously-supported domain-scoped project IDs,
+    // which were in the format of <DOMAIN>.<PROJECT>. The domain can contain ":".
+    let matchFullBillingTableID = "/^[a-z][a-z-\d]{5,29}\.[a-zA-Z_\d]{1,1024}\.[a-zA-Z_\d]{1,1024}$/"
+    let matchDomainScopedBillingTableID = "/^[a-z][a-z-\d]+\.*[a-z][a-z-\d]+:[a-z][a-z-\d]+\.[a-zA-Z_\d]{1,1024}\.[a-zA-Z_\d]{1,1024}$/"
     // fullBillingTable should be in the format of ${PROJECT_ID}.${DATASET_ID}.${TABLE_ID}.
     // Bigquery table IDs can contain letters, numbers and underscores.
-    if (!fullBillingTable.match(/^[a-z][a-z-\d]{5,29}\.[a-zA-Z_\d]{1,1024}\.[a-zA-Z_\d]{1,1024}$/)) {
+    if
+      (!fullBillingTable.match(matchFullBillingTableID)
+      && !fullBillingTable.match(matchDomainScopedBillingTableID)) {
```
Comment (fabio-rigato) [inline gke-usage-metering/src/gkeUsageMetering.ts:162]: I think we need an OR instead of AND @sarandia

**Pull request merged**